jsettlers.graphics/src/main/java/jsettlers/graphics/map/draw/settlerimages/SettlerImageMapItem.java

```diff
@@ -52,4 +52,31 @@ public int getStart() {
 	public int getDuration() {
 		return this.duration;
 	}
+
+	@Override
+	public boolean equals(Object o) {
+		if (this == o) return true;
+		if (o == null || getClass() != o.getClass()) return false;
+		SettlerImageMapItem that = (SettlerImageMapItem) o;
+		return file == that.file &&
+				sequenceIndex == that.sequenceIndex &&
+				start == that.start &&
+				duration == that.duration;
+	}
+
+	@Override
+	public int hashCode() {
+		return Objects.hash(file, sequenceIndex, start, duration);
```

Comment:
Same allocation problem here - use direct hash code computation instead to avoid allocating an array.

Reply:
This object is used as a value in the hash map so no hashing is done. I don't know why I added hashCode and equals here ^^